src/distPlus/symbolic/SymbolicDist.re

```diff
@@ -296,103 +313,407 @@ module GenericSimple = {
+        let n: normal = {mean: n1.mean -. n2.mean, stdev: sqrt(n1.stdev ** 2. +. n2.stdev ** 2.)};
+        `Simple(`Normal(n));
+      }
+
+    | (`Simple(`Lognormal(l1)), `Simple(`Lognormal(l2)), `MultiplyOperation) => {
+        let l: lognormal = {mu: l1.mu +. l2.mu, sigma: l1.sigma +. l2.sigma};
+        `Simple(`Lognormal(l));
+      }
+
+    | (`Simple(`Lognormal(l1)), `Simple(`Lognormal(l2)), `DivideOperation) => {
+        let l: lognormal = {mu: l1.mu -. l2.mu, sigma: l1.sigma +. l2.sigma};
+        `Simple(`Lognormal(l));
+      }
+
+
+    /* General cases: convolve the XYShapes */
+    | (`Simple(d1), `Simple(d2), _) => {
+        let (sc1, sd1) = renderDistributionToXYShape(d1, n);
+        let (sc2, sd2) = renderDistributionToXYShape(d2, n);
+        let (sc, sd) = combinationDistributionOfXYShapes(sc1, sd1, sc2, sd2, func);
+        `RenderedShape(sc, sd, 1.0)
+    }
+    | (`Simple(d1), `RenderedShape(sc2, sd2, i2), _)
+    | (`RenderedShape(sc2, sd2, i2), `Simple(d1), _) => {
+        let (sc1, sd1) = renderDistributionToXYShape(d1, n);
+        let (sc, sd) = combinationDistributionOfXYShapes(sc1, sd1, sc2, sd2, func);
+        `RenderedShape(sc, sd, i2)
+    }
+    | (`RenderedShape(sc1, sd1, i1), `RenderedShape(sc2, sd2, i2), _) => {
+        // sum of two multimodals that have a continuous and discrete each.
+        let (sc, sd) = combinationDistributionOfXYShapes(sc1, sd1, sc2, sd2, func);
+        `RenderedShape(sc, sd, i1);
+      }
+    }
   };
 
-  let toString = (dists: t) => {
-    let distString =
-      dists
-      |> E.A.fmap(d => GenericSimple.toString(fst(d)))
-      |> Js.Array.joinWith(",");
-    let weights =
-      dists
-      |> E.A.fmap(d =>
-           snd(d) |> Js.Float.toPrecisionWithPrecision(~digits=2)
-         )
-      |> Js.Array.joinWith(",");
-    {j|multimodal($distString, [$weights])|j};
+  let evaluatePointwiseSum = (et1: nodeResult, et2: nodeResult, n: int) => {
+    switch ((et1, et2)) {
+    /* Known cases: */
+    | (`Simple(`Float(v1)), `Simple(`Float(v2))) => {
+        v1 == v2
+        ? `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.make({xs: [|v1|], ys: [|2.|]}), 2.)
+        : `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.) // TODO: add warning: shouldn't pointwise add scalars.
+    }
+    | (`Simple(`Float(v1)), `Simple(d2))
+    | (`Simple(d2), `Simple(`Float(v1))) => {
+        let sd1: DistTypes.xyShape = {xs: [|v1|], ys: [|1.|]};
+        let (sc2, sd2) = renderDistributionToXYShape(d2, n);
+        `RenderedShape(sc2, Distributions.Discrete.reduce((+.), [|sd1, sd2|]), 2.)
+    }
+    | (`Simple(d1), `Simple(d2)) => {
+        let (sc1, sd1) = renderDistributionToXYShape(d1, n);
+        let (sc2, sd2) = renderDistributionToXYShape(d2, n);
+        `RenderedShape(Distributions.Continuous.reduce((+.), [|sc1, sc2|]), Distributions.Discrete.reduce((+.), [|sd1, sd2|]), 2.)
```

Comment:
Have you spent much time looking through ``Distributions``? You could make this into a mixed distribution, then use that to combine the two. Really what's going on here is that there's a type of ``Mixed`` distribution, and Mixed distributions can be added to each other.

Reply:
> Have you spent much time looking through `Distributions`?

Evidently not :) I'll have another look!

Reply:
It's possible there's not great code yet to reduce Mixed distributions; but that should be the strategy here imo.